python_modules/dagster/dagster/_daemon/sensor.py

```diff
@@ -451,16 +533,8 @@ def _process_tick_generator(
         mark_sensor_state_for_tick(instance, external_sensor, sensor_state, now)
 
     try:
-        tick = instance.create_tick(
-            TickData(
-                instigator_origin_id=sensor_state.instigator_origin_id,
-                instigator_name=sensor_state.instigator_name,
-                instigator_type=InstigatorType.SENSOR,
-                status=TickStatus.STARTED,
-                timestamp=now.timestamp(),
-                selector_id=external_sensor.selector_id,
-            )
-        )
+        # get the tick that we should be evaluating for
+        tick = _get_evaluation_tick(instance, external_sensor, now.timestamp(), logger)
```

Comment:
One potentially significant change here is that this now results in each sensor doing an additional fetch on each iteration to check over the most recent tick. The scheduler daemon and asset daemon already do this, but schedules are typically no more than once an hour and most orgs only have a single AMP 'sensor', this would be for every sensor every 30 seconds. @alangenfeld you may have thoughts here since you have been digging into high-throughput sensor perf recently

Reply:
(That said, we were already doing a write every 30 seconds anyway to create new ticks at the same cadence, so its also possible this is fine?)

Reply:
but if we were concerned about this, we could consider mitigations like:
- only doing this new retry behavior for asset sensors
- Making it opt-in in some way by applying a retry policy to specific sensors
- Making it opt-out and adding the ability to turn it off in high-throughput situations
- Making it opt-in and rolling it out more gradually

Reply:
hmm hard for me to guess what impact an additional tick fetch would have on throughput. 

One idea that pops to mind is to use `SensorInstigatorData` since we already have that fetched and store other `last_tick_...` fields on it. Maybe not all the data there but enough to only have to fetch the last tick in the uncommon recovery scenario

Reply:
@alangenfeld good idea -- I updated the PR to add a `last_tick_end_timestamp` which gets set to `None` when the tick starts, and set to a float value if the tick is able to gracefully exit.

In cases where we start an evaluation and that is set to `None`, then we can query for the previous tick to get the tick that was interrupted and go from there, but otherwise (the common case), we can just proceed as normal.

Technically, we don't use the timestamp value at all, and so I could have just made this a "last tick ended gracefully" boolean, but chose this approach as it seemed more general (maybe we come up with a use for this in the future), and had nice symmetry with `last_tick_start_timestamp`.